### PR TITLE
[TGBOT-292] Engineer-methods

### PR DESCRIPTION
**Pull request by AlexanderYarokhovich**
https://schneidervs.atlassian.net/jira/software/projects/TGBOT/boards/2?selectedIssue=TGBOT-292

### CODE REVIEW

**Review by schneidervs — APPROVED** [review]

**Review by Artem-Metlushko — DISMISSED** [review]

File: web-ui/src/test/java/com/tutrit/webclient/controller/EngineerControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    }
+
+    private Engineer makeEngineer() {
+        return new Engineer("id1", "firstName1", "lastName1", "function1", "category1", "education1", 3, 10);
+    }
+
+    private Engineer verifyEngineer() {
+        return new Engineer("52", "firstName1", "lastName1", "function1", "category1", "education1", 3, 10);
+    }
+}
```
Comment (schneidervs) [inline web-ui/src/test/java/com/tutrit/webclient/controller/EngineerControllerTest.java]: GitHub says no empty string

**Review by schneidervs — DISMISSED** [review]

**Review by yurasava — DISMISSED** [review]

File: web-core/src/main/java/com/tutrit/gateway/EngineerGateway.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package com.tutrit.gateway;
+
+import com.tutrit.bean.Engineer;
+
+public interface EngineerGateway {
+    Engineer saveEngineer(Engineer engineer);
+    Engineer findEngineerById(String id);
```
Comment (max-java) [inline web-core/src/main/java/com/tutrit/gateway/EngineerGateway.java]: Maybe use Optional here?

File: web-ui/src/main/java/com/tutrit/webclient/controller/EngineerController.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import com.tutrit.gateway.EngineerGateway;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.servlet.ModelAndView;
+
+import java.util.Optional;
+
+@Controller
+//@RequestMapping("/engineers")
```
Comment (max-java) [inline web-ui/src/main/java/com/tutrit/webclient/controller/EngineerController.java]: Either uncomment or remove comment

File: web-ui/src/test/java/com/tutrit/config/MySpringContext.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
+package com.tutrit.config;
+
+import org.springframework.boot.SpringBootConfiguration;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+
+public class MySpringContext {
```
Comment (max-java) [inline web-ui/src/test/java/com/tutrit/config/MySpringContext.java]: Let's rename it to SpringContexConfig

File: web-ui/src/test/java/com/tutrit/webclient/controller/EngineerControllerTest.java (hunk 1 above)
Comment (max-java) [inline web-ui/src/test/java/com/tutrit/webclient/controller/EngineerControllerTest.java]: add new line

**Review by max-java — CHANGES_REQUESTED** [review]
ptal

File: web-ui/src/test/java/com/tutrit/webclient/controller/EngineerControllerTest.java (hunk 1 above)
Comment (max-java) [reply web-ui/src/test/java/com/tutrit/webclient/controller/EngineerControllerTest.java]: GitHub says no empty string = github complains on empty string absence

**Comment by max-java** [comment]
Any updates on this?

**Comment by AlexanderYarokhovich** [comment]
> Есть новости по этому поводу?

File: web-ui/src/test/java/com/tutrit/webclient/controller/EngineerControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+@AutoConfigureMockMvc
+@SpringBootTest(classes = SpringContextConfig.SpringConfig.class)
+class EngineerControllerTest {
+
+    @MockBean
+    EngineerGateway engineerGateway;
+    @Autowired
+    MockMvc mockMvc;
+    @Autowired
+    ObjectMapper objectMapper;
```
Comment (yurasava) [inline web-ui/src/test/java/com/tutrit/webclient/controller/EngineerControllerTest.java]: It is not used in these tests

File: web-ui/src/test/java/com/tutrit/webclient/controller/EngineerControllerTest.java [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+        final MvcResult mvcResult = mockMvc
+                .perform(MockMvcRequestBuilders.post("/engineers/52?" +
+                                "engineerId=52" +
+                                "&firstName=firstName1" +
+                                "&lastName=lastName1" +
+                                "&function=function1" +
+                                "&category=category1" +
+                                "&education=education1" +
+                                "&experience=3" +
+                                "&generalExperience=10")
```
Comment (yurasava) [inline web-ui/src/test/java/com/tutrit/webclient/controller/EngineerControllerTest.java]: You can put all keys and values as parameters as in deleteEngineer() test

**Review by yurasava — DISMISSED** [review]

**Review by Artem-Metlushko — DISMISSED** [review]

File: web-ui/src/test/java/com/tutrit/webclient/controller/EngineerControllerTest.java (hunk 6 above)
Comment (AlexanderYarokhovich) [reply web-ui/src/test/java/com/tutrit/webclient/controller/EngineerControllerTest.java]: done

File: web-ui/src/test/java/com/tutrit/webclient/controller/EngineerControllerTest.java (hunk 5 above)
Comment (AlexanderYarokhovich) [reply web-ui/src/test/java/com/tutrit/webclient/controller/EngineerControllerTest.java]: done

**Review by schneidervs — APPROVED** [review]

**Review by Sav4ykVladzislau — APPROVED** [review]
lgfm

**Review by yurasava — APPROVED** [review]

**Review by Artem-Metlushko — APPROVED** [review]

**Review by max-java — APPROVED** [review]
lgfm

**Pull request merged**